cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -383,16 +388,51 @@ func ValidateIPNetFromString(subnet string, minAddrs int64, fldPath *field.Path)
 	return allErrs
 }
 
-// ValidateNetworking validates networking configuration
+// ValidateIPNetFromStringForDualStack validates network portion of IP addresses for dual-stack configurations
+func ValidateIPNetFromStringForDualStack(subnet string, minAddrs int64, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	subnets := strings.Split(subnet, ",")
+	if len(subnets) < 2 {
+		allErrs = append(allErrs, field.Invalid(fldPath, subnet, "expected more than 1 pod subnet in dual-stack configuration"))
+	}
+	for _, subNetStr := range subnets {
+		_, subNet, err := net.ParseCIDR(strings.TrimSpace(subNetStr))
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, subNetStr, "couldn't parse subnet"))
+		} else {
+			numAddresses := ipallocator.RangeSize(subNet)
+			if numAddresses < minAddrs {
+				allErrs = append(allErrs, field.Invalid(fldPath, subnet, "subnet is too small"))
+			}
+		}
+	}
+	return allErrs
+}
+
+// isDualStackNetworking returns true if more than one pod CIDR has been specified
+func isDualStackNetworking(c *kubeadm.Networking) bool {
```

Comment:
I think that the conditions is that it has more than one CIDR , and each CIDR belongs to a different IP family. Maybe we can leverage these functions in k/utils https://github.com/kubernetes/utils/blob/581e00157fb1a0435d4fac54a52d1ca1e481d60e/net/net.go#L104

Reply:
I'm +1 on checking more than one CIDR , and each CIDR belongs to a different IP family

Reply:
Done!